### PR TITLE
[GITHUB-11] Switches system integration to infra

### PR DESCRIPTION
**Pull request by lobsterdore**
New Relic have deprecated Sysmond in favour of their Infra
integration, this change swithes this role over.

Additionally changes:

* Modified labels to be a dictionary so it is a little
bit more flexiblr
* Added displayname var which ensures that both Infra and PHP
integrations use the same host name
* Makes repo keys and URLs configurable

### CODE REVIEW

**Comment by lobsterdore** [comment]
Will close #11

File: defaults/main.yml [hunk 1]
```diff
@@ -2,8 +2,25 @@
...
+  - 548C16BF
+sansible_newrelic_apt_repos:
+  - "deb [arch=amd64] https://download.newrelic.com/infrastructure_agent/linux/apt trusty main"
+  - "deb http://apt.newrelic.com/debian/ newrelic non-free"
+
 sansible_newrelic_group: newrelic
 
+sansible_newrelic_integrations_infra_display_name: "{{ ansible_hostname }}"
+sansible_newrelic_integrations_infra_enabled: false
+sansible_newrelic_integrations_infra_ini_config: [ ]
```
Comment (dennisconrad) [inline defaults/main.yml]: No space in `[]`

File: defaults/main.yml [hunk 2]
```diff
@@ -17,16 +34,8 @@ sansible_newrelic_integrations_php_ini_config: [ ]
...
-  # - option: number_of_cats
-  #   section: mammals
-  #   value: 42
-sansible_newrelic_integrations_sysmond_start_on_boot: no
-sansible_newrelic_integrations_sysmond_version: "2.3.*"
-
-# Format is: key:value,key:value
-sansible_newrelic_labels: ~
+# Hash format: sansible_newrelic_labels: { key: value, key: value }
+sansible_newrelic_labels: { }
```
Comment (dennisconrad) [inline defaults/main.yml]: No space in `{}`

File: README.md [hunk 3]
```diff
@@ -78,17 +78,19 @@ during OS boot.
...
 ```YAML
 - name: Install and configure newrelic
   hosts: "somehost"
 
   roles:
     - role: sansible.newrelic
-      sansible_newrelic_integrations_sysmond_enabled: true
-      sansible_newrelic_integrations_sysmond_start_on_boot: true
-      sansible_newrelic_labels: "org:sansible,role:newrelic"
+      sansible_newrelic_integrations_infra_enabled: true
```
Comment (dennisconrad) [inline README.md]: `yes`/ `no` instead of `true` / `false` is better.

**Review by rsgoodall — APPROVED** [review]

File: tasks/configure.yml [hunk 4]
```diff
@@ -74,10 +56,32 @@
...
+    option: newrelic.labels
+    value: "\"{%- set tmp_list = [] -%}
+              {%- for attribute_key, attribute_value in sansible_newrelic_labels.iteritems() -%}
+                {{ tmp_list.append(attribute_key + ':' + attribute_value) }}
+              {%- endfor -%}
+            {{ tmp_list |join(';') }}\""
+  when:
+    - sansible_newrelic_integrations_php_enabled
+    - sansible_newrelic_integrations_php_start_on_boot
+    - 'sansible_newrelic_labels | length'
```
Comment (dennisconrad) [inline tasks/configure.yml]: I would use `sansible_newrelic_labels | length > 0`

**Review by dennisconrad — APPROVED** [review]
👍

**Pull request merged**